modules/core/src/main/java/org/apache/synapse/config/SynapseConfiguration.java

```diff
@@ -525,6 +551,18 @@ public synchronized void removeAPI(String name) {
         }
     }
 
+    private void removeBindsToMappings(String apiName) {
+        Iterator<Map.Entry<String, Map<String, API>>> iterator = apiTableWithBindsTo.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, Map<String, API>> mappings = iterator.next();
+            Map<String, API> apis = mappings.getValue();
+            apis.remove(apiName);
```

Comment:
Can apis null here?

Reply:
No, since we don't put `null` valued `API`s to this map.